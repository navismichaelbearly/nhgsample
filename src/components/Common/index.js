import styled from 'styled-components';
import { Button, mediaQueries, Typography } from 'nhh-styles';

/**
 * file to aggregate styled components and hopefully allow more reuse of them
 * also to eventually replace all the labels used as heading etc
 */
export const Heading = styled(Typography.H3)`
  margin-bottom: 1.5rem;
`;

export const Label = styled.span`
  font-weight: 700;
  ${mediaQueries.desktop`
    white-space: nowrap;
  `};
  ${mediaQueries.tablet`
    white-space: nowrap;
  `};
`;

export const LabelLink = styled(Button)`
  font-weight: 700;
  font-size: 18px;
  padding: 0;
  text-align: left;
`;

export const Wrapper = styled.div`
  margin-bottom: 1.5rem;
`;

export const NotificationWrapper = styled.div`
  margin-bottom: 30px;
`;
