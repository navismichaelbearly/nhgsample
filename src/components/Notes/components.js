import styled from 'styled-components';
import { Typography } from 'nhh-styles';

export const NoteItem = styled.div`
  margin-bottom: 1rem;
`;

export const NoteBody = styled.div`
  word-break: break-all;
`;

export const NoteLabel = styled.span`
  font-weight: 700;
`;

export const NoteDescription = styled.p`
  margin-top: 1rem;
`;

export const Wrapper = styled.div`
  clear: both;
  margin-bottom: 1.5rem;
  padding-top: 1rem;
`;

export const H3 = styled(Typography.H3)`
  margin-bottom: 30px;
`;
