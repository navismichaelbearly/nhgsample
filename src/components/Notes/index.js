import { Note, NotePropType, NoteDictionaryPropType } from './Note';

export { default as NoteList } from './NoteList';
export { Note, NotePropType, NoteDictionaryPropType };
