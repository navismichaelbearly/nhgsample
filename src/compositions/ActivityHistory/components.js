import styled from 'styled-components';
import { Button, Typography } from 'nhh-styles';

export const Activity = styled.div`
  margin-bottom: 1rem;
`;

export const ActivityBody = styled.div``;

export const ActivityLabel = styled.span`
  font-weight: 700;
`;

export const ActivityLink = styled(Button)`
  font-size: inherit;
`;

export const H4 = styled(Typography.H4)`
  font-size: 1em;
  margin-bottom: 0.7rem;
`;

export const Wrapper = styled.div`
  clear: both;
  margin-bottom: 1.5rem;
  padding-top: 1rem;
`;

export const H3 = styled(Typography.H3)`
  margin-bottom: 30px;
`;

export const ActivityDescription = styled.span`
  word-break: break-word;
`;
