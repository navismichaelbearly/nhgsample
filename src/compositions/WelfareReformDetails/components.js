import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Row = styled.div`
  margin-bottom: 15px;
`;

export const FootNote = styled.div`
  border: 1px;
  color: #aaa;
  font-size: 14px;
`;
