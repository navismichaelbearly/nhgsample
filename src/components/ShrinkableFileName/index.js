import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'nhh-styles';
import styled from 'styled-components';
import { shortenFileName } from '../../util/splitAndWrap';
import { LabelWrapper } from './components';

const shorterNameForDesktop = shortenFileName(26);
const shorterNameForMobile = shortenFileName(7);

const AttachmentsTableFileName = ({ name, isLink, onClick }) => {
  const ParentElement = isLink ? Button : styled.span``;
  const parentProps = isLink ? { isText: true, onClick } : {};

  return (
    <LabelWrapper>
      <ParentElement {...parentProps}>
        <span className="d-none d-lg-block">{shorterNameForDesktop(name)}</span>
        <span className="d-lg-none">{shorterNameForMobile(name)}</span>
      </ParentElement>
    </LabelWrapper>
  );
};

AttachmentsTableFileName.defaultProps = {
  isLink: false,
  onClick: () => {},
};

AttachmentsTableFileName.propTypes = {
  name: PropTypes.string.isRequired,
  isLink: PropTypes.bool,
  onClick: PropTypes.func,
};

export default AttachmentsTableFileName;
