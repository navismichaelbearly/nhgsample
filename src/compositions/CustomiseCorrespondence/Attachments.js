import React from 'react';
import PropTypes from 'prop-types';
import { AdvancedFileUpload, Table, Typography, Button, Pagination } from 'nhh-styles';
import { MediaTableView } from '../../components';
import { MandatoryField, TableWrapper, AttachmentsWrapper, ErrorText } from './components';
import AttachmentsTableFileName from '../../components/ShrinkableFileName';

class Attachments extends React.Component {
  state = {
    pageSize: 6,
  };

  componentDidMount() {
    this.props.getFilesByCaseId();
  }
  componentWillUnmount() {
    this.props.clearMediaState();
  }

  addCaseFile = fileUri => this.props.handleFilesChange([...this.props.selectedCaseFiles, fileUri]);

  removeCaseFile = fileUri =>
    this.props.handleFilesChange(this.props.selectedCaseFiles.filter(x => x !== fileUri));

  buildCaseFilesTable = () => {
    const { mandatoryAttachments, downloadFile, downloadedFiles } = this.props;

    if (!mandatoryAttachments.length && !downloadedFiles.length) return null;

    const mandatoryFilesTableData = mandatoryAttachments.map(file => [
      <AttachmentsTableFileName name={file.friendlyName} />,
      <MandatoryField>{this.props.dictionary.mandatoryAttachment}</MandatoryField>,
    ]);

    const caseFilesTableData = downloadedFiles.map(file => {
      const isSelected = this.props.selectedCaseFiles.includes(file.uri);

      return [
        <AttachmentsTableFileName
          isLink
          onClick={() => downloadFile(file.uri, file.name)}
          name={file.name}
        />,
        isSelected ? (
          <Button isText onClick={() => this.removeCaseFile(file.uri)}>
            {this.props.dictionary.removeCaseFile}
          </Button>
        ) : (
          <Button className={'success'} isText onClick={() => this.addCaseFile(file.uri)}>
            {this.props.dictionary.attachCaseFile}
          </Button>
        ),
      ];
    });

    const { pageSize } = this.state;

    return (
      <AttachmentsWrapper>
        <Typography.Label>{this.props.dictionary.mandatoryAttachmentLabel}</Typography.Label>
        <Typography.P>{this.props.dictionary.mandatoryAttachmentParagraph}</Typography.P>
        <TableWrapper>
          <Pagination
            initialPageSize={pageSize}
            items={mandatoryFilesTableData.concat(caseFilesTableData)}
            pageSize={6}
            render={items => {
              if (items.length > pageSize) {
                this.setState({ pageSize: items.length });
              }

              return <Table addExtraTd={false} tdWrap data={items} />;
            }}
          />
        </TableWrapper>
      </AttachmentsWrapper>
    );
  };

  render() {
    if (!this.props.hasGeneratedCorrespondence) {
      return null;
    }

    const caseFilesTable = this.buildCaseFilesTable();

    const mediaTableViewProps = {
      // for clarity only; 'files' is passed automatically to renderer by AdvancedFileUpload
      files: this.props.pendingFiles,
      // for clarity only 'onRemove' is passed automatically to renderer by AdvancedFileUpload
      onRemove: this.props.onRemove,
      onRetry: this.props.onRetry,
      removeButtonLabel: this.props.dictionary.removeButtonLabel,
      retryButtonLabel: this.props.dictionary.retryButtonLabel,
      uploadSuccessMessage: this.props.dictionary.uploadSuccessMessage,
      genericErrorMessage: this.props.dictionary.genericErrorMessage,
    };

    return (
      <React.Fragment>
        {caseFilesTable}
        {this.props.hasDownloadError && (
          <ErrorText>{this.props.dictionary.genericErrorMessage}</ErrorText>
        )}
        <AttachmentsWrapper>
          <AdvancedFileUpload
            {...this.props.dictionary}
            accept={this.props.acceptedFileExtensions}
            files={this.props.pendingFiles}
            labelText={this.props.dictionary.labelText}
            maxFiles={this.props.maxFiles}
            maxFileSize={this.props.maxFileSize}
            onChange={this.props.onChange}
            onRemove={this.props.onRemove}
            previewsRenderer={MediaTableView}
            text={this.props.dictionary.text}
            {...mediaTableViewProps}
          />
        </AttachmentsWrapper>
      </React.Fragment>
    );
  }
}

Attachments.defaultProps = {
  dictionary: {},
  hasDownloadError: false,
};

export const attachmentPropTypes = {
  acceptedFileExtensions: PropTypes.string.isRequired,
  clearMediaState: PropTypes.func.isRequired,
  dictionary: PropTypes.shape({
    attachCaseFile: PropTypes.string.isRequired,
    genericErrorMessage: PropTypes.string.isRequired,
    labelText: PropTypes.string.isRequired,
    mandatoryAttachment: PropTypes.string.isRequired,
    mandatoryAttachmentLabel: PropTypes.string.isRequired,
    mandatoryAttachmentParagraph: PropTypes.string.isRequired,
    removeButtonLabel: PropTypes.string.isRequired,
    removeCaseFile: PropTypes.string.isRequired,
    retryButtonLabel: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    uploadSuccessMessage: PropTypes.string.isRequired,
  }).isRequired,
  downloadedFiles: PropTypes.array.isRequired,
  downloadFile: PropTypes.func.isRequired,
  getFilesByCaseId: PropTypes.func.isRequired,
  handleFilesChange: PropTypes.func.isRequired,
  hasGeneratedCorrespondence: PropTypes.bool.isRequired,
  mandatoryAttachments: PropTypes.array.isRequired,
  maxFiles: PropTypes.number.isRequired,
  maxFileSize: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  onRetry: PropTypes.func.isRequired,
  pendingFiles: PropTypes.arrayOf(
    PropTypes.shape({
      errorText: PropTypes.string,
      hasBeenUploaded: PropTypes.bool,
      hasUploadError: PropTypes.bool,
      id: PropTypes.string,
      isUploading: PropTypes.bool,
    })
  ).isRequired,
  selectedCaseFiles: PropTypes.arrayOf(PropTypes.string).isRequired,
  hasDownloadError: PropTypes.bool,
};

Attachments.propTypes = attachmentPropTypes;

export default Attachments;
