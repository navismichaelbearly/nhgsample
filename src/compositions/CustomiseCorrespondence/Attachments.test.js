import React from 'react';
import { shallow, mount } from 'enzyme';
import { testUtils, Pagination } from 'nhh-styles';
import Attachments from './Attachments';
import { TableWrapper, ErrorText } from './components';

describe('<Attachments />', () => {
  let props;
  let el;

  beforeEach(() => {
    props = {
      acceptedFileExtensions: '.doc,.docx,.xls,.xlsx,.ppt,.pptx,.pdf,.txt,.jpg,.png',
      downloadFile: jest.fn(),
      selectedCaseFiles: [],
      handleFilesChange: jest.fn(),
      clearMediaState: jest.fn(),
      downloadedFiles: [],
      getFilesByCaseId: jest.fn(),
      hasGeneratedCorrespondence: true,
      hasDownloadError: false,
      mandatoryAttachments: [],
      maxFileSize: 20971520,
      maxFiles: 10,
      onChange: jest.fn(),
      onRemove: jest.fn(),
      onRetry: jest.fn(),
      pendingFiles: [],
      dictionary: {
        attachCaseFile: 'attachCaseFileSample',
        genericErrorMessage: 'Some awesome error',
        labelText: 'labelTextSample',
        mandatoryAttachment: 'This is mandatory',
        mandatoryAttachmentLabel: 'Mandatory label',
        mandatoryAttachmentParagraph: 'This paragraph is cool',
        removeButtonLabel: 'remove this',
        removeCaseFile: 'removeCaseFileSample',
        retryButtonLabel: 'retry this',
        text: 'textSample',
        uploadSuccessMessage: 'well done',
      },
    };

    el = shallow(<Attachments {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should not render when correspondence has not been generated yet', () => {
    props.hasGeneratedCorrespondence = false;
    expect(shallow(<Attachments {...props} />)).toMatchSnapshot();
  });

  it('should render correctly when correspondence has been generated', () => {
    expect(el).toMatchSnapshot();
  });

  it('should call getFilesByCaseId on mount', () => {
    expect(props.getFilesByCaseId).toHaveBeenCalled();
  });

  it('should clear media state on unmount', () => {
    el.unmount();
    expect(props.clearMediaState).toHaveBeenCalled();
  });

  it('should show error when there is a download error', () => {
    props = {
      ...props,
      hasDownloadError: true,
    };

    const mounted = mount(testUtils.withTheme(<Attachments {...props} />));
    expect(mounted.find(ErrorText)).toExist();
    expect(mounted.find(ErrorText).text()).toBe(props.dictionary.genericErrorMessage);
  });

  describe('with case files', () => {
    let newEl;

    beforeEach(() => {
      props = {
        ...props,
        selectedCaseFiles: ['thisIsAFileURI'],
        mandatoryAttachments: [{ friendlyName: 'hello' }],
        downloadedFiles: [
          { name: 'hello world', uri: '1234' },
          { name: 'hello new world', uri: '221' },
        ],
      };

      newEl = shallow(<Attachments {...props} />);
    });

    it('should update local pageSize state when pagination amends items after view more click', () => {
      newEl.setState({ pageSize: 1 });
      expect(newEl.state().pageSize).toBe(1);
      const paginationProps = newEl.find(Pagination).props();
      paginationProps.render(paginationProps.items.slice(0, 2));
      expect(newEl.state().pageSize).toBe(2);
    });

    it('should render case files table when there are mandatory files or downloaded files available', () => {
      expect(el.find(TableWrapper)).not.toExist();
      expect(newEl.find(TableWrapper)).toExist();
    });

    it('should add optional case files to the state', () => {
      expect(props.selectedCaseFiles.length).toBe(1);
      newEl.instance().addCaseFile('someCaseFile');
      expect(props.handleFilesChange).toHaveBeenCalledWith(['thisIsAFileURI', 'someCaseFile']);
    });

    it('should remove optional case files from state', () => {
      expect(props.selectedCaseFiles.length).toBe(1);
      newEl.instance().removeCaseFile('thisIsAFileURI');
      expect(props.handleFilesChange).toHaveBeenCalledWith([]);
    });
  });
});
