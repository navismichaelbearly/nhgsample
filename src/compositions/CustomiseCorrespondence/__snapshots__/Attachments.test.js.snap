// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Attachments /> should not render when correspondence has not been generated yet 1`] = `""`;

exports[`<Attachments /> should render correctly when correspondence has been generated 1`] = `
<Fragment>
  <styled.div>
    <t
      accept=".doc,.docx,.xls,.xlsx,.ppt,.pptx,.pdf,.txt,.jpg,.png"
      attachCaseFile="attachCaseFileSample"
      files={Array []}
      genericErrorMessage="Some awesome error"
      labelText="labelTextSample"
      mandatoryAttachment="This is mandatory"
      mandatoryAttachmentLabel="Mandatory label"
      mandatoryAttachmentParagraph="This paragraph is cool"
      maxFileSize={20971520}
      maxFiles={10}
      multiple={true}
      onChange={[Function]}
      onRemove={[Function]}
      onRetry={[Function]}
      previewsRenderer={[Function]}
      removeButtonLabel="remove this"
      removeCaseFile="removeCaseFileSample"
      retryButtonLabel="retry this"
      text="textSample"
      uploadSuccessMessage="well done"
    />
  </styled.div>
</Fragment>
`;
